qiita_db/test/test_metadata_template.py

```diff
@@ -1363,6 +1364,42 @@ def test_create_bad_sample_names(self):
             PrepTemplate.create(self.metadata, self.new_raw_data,
                                 self.test_study, self.data_type)
 
+    def test_create_unknown_sample_names(self):
+        # set two real and one fake sample name
+        self.metadata_dict['NOTREAL'] = self.metadata_dict['SKB7.640196']
+        del self.metadata_dict['SKB7.640196']
+        self.metadata = pd.DataFrame.from_dict(self.metadata_dict,
+                                               orient='index')
+        # Test error raised and correct error given
+        with self.assertRaises(QiitaDBExecutionError):
```

Comment:
You can also move the assertEqual outside the context manager.

On (Feb-25-15|10:14), Joshua Shorenstein wrote:
> > @@ -1363,6 +1364,42 @@ def test_create_bad_sample_names(self):
> >              PrepTemplate.create(self.metadata, self.new_raw_data,
> >                                  self.test_study, self.data_type)
> >  
> > +    def test_create_unknown_sample_names(self):
> > +        # set two real and one fake sample name
> > +        self.metadata_dict['NOTREAL'] = self.metadata_dict['SKB7.640196']
> > +        del self.metadata_dict['SKB7.640196']
> > +        self.metadata = pd.DataFrame.from_dict(self.metadata_dict,
> > +                                               orient='index')
> > +        # Test error raised and correct error given
> > +        with self.assertRaises(QiitaDBExecutionError):
> 
> I had it that way originally, but that only works with warnings. Anything inside the with after the error is raised is ignored.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/biocore/qiita/pull/870/files#r25363609